coordinator-tool/src/bin/ec.rs

```diff
@@ -0,0 +1,895 @@
+                    };
+                    let Some(share) = decryption_share.get(i) else {
+                        anyhow::bail!("Missing decryption share for contest {contest} and option {i}");
+                    };
+                    decryption_shares_for_option.push(share);
+                }
+                let mut proof_shares_for_option = Vec::new();
+                for proof_shares in &proof_shares {
+                    let Some(proof_share) = proof_shares.get(&contest) else {
+                        anyhow::bail!("Missing proof share for contest {contest}");
+                    };
+                    let Some(share) = proof_share.get(i) else {
+                        anyhow::bail!("Missing proof share for contest {contest} and option {i}");
+                    };
+                    proof_shares_for_option.push(share);
+                }
+
+                let decrypted = VerifiableDecryption::compute(
+                    &election_data.manifest,
+                    &election_data.parameters,
+                    &election_data.guardian_public_keys,
+                    &ciphertext,
+                    decryption_shares_for_option,
+                    proof_shares_for_option,
+                )?;
+                ciphers.push(decrypted);
+            }
+            decrypted_tallies.insert(contest, ciphers);
+        }
+        decrypted_tallies
+    };
+
+    let contest = {
+        let mut contests = election_data.manifest.contests.indices();
+        let Some(contest) = contests.next() else {
+            anyhow::bail!("Need a contest in manifest.");
+        };
+        anyhow::ensure!(
+            contests.next().is_none(),
+            "Only a single contest is supported."
+        );
+        contest
+    };
+    let Some(results) = decryption.remove(&contest) else {
+        anyhow::bail!("No decryptions for contest.");
+    };
+
+    let mut weights: contract::PostResultParameter = Vec::with_capacity(results.len());
+    for value in results {
+        let weight = value.plain_text.to_u64_digits();
+        eprintln!("{weight:?}");
+        anyhow::ensure!(weight.len() <= 1, "Weight must fit into a u64.");
+        weights.push(weight.first().copied().unwrap_or(0));
+    }
+
+    let current_result = contract_client
+        .view::<_, contract::ViewElectionResultQueryResponse, ViewError>(
+            "viewElectionResult",
+            &(),
+            BlockIdentifier::LastFinal,
+        )
+        .await?;
+
+    if let Some(result) = current_result {
+        let current_weights = result
+            .iter()
+            .map(|x| x.cummulative_votes)
+            .collect::<Vec<_>>();
+        if current_weights != weights {
+            // TODO: Should we allow uploading again?
+            anyhow::bail!(
+                "The election results are already registered in the contract, but they are \
+                 different."
+            );
```

Comment:
Is there a reason why we would not allow this? The expected case is that if run multiple times, the script will exit due to the results matching. In the case they are not something unexpected will have happened in which case one might want to investigate?

Would it make sense to ask for confirmation to override the currently registered result?

Reply:
I have added a prompt now.

There should not really be a case where this happens. And I was considering disallowing it in the contract as well.But perhaps the added flexibility is useful.

Reply:
This is allowed now with a prompt.